sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1593,6 +1593,35 @@ class Dataset[T] private[sql](
   @scala.annotation.varargs
   def select(col: String, cols: String*): DataFrame = select((col +: cols).map(Column(_)) : _*)
 
+  /**
+   * Selects a set of columns via schema object.
+   */
+  def select(schema: StructType): DataFrame = {
+    val attrs = logicalPlan.output
+    val attrs_map = attrs.map { a => (a.name, a) }.toMap
+    val new_attrs = AttributeMap(schema.map { f =>
```

Comment:
We will have to make it working with nested fields I think ..

Reply:
So this current code swaps the `dataType` without checking if it's safe. Do you or @jiangxb1987 have any tips how to safely handle nested data here?

Reply:
👋 @HyukjinKwon @jiangxb1987 would appreciate your input please ^

Reply:
@HyukjinKwon @jiangxb1987 would appreciate your input please ^

Reply:
@HyukjinKwon @jiangxb1987 one thing we could do is to validate that the schema we are selecting/swapping by is compatible with the current schema (including nested columns). Would this be an acceptable solution for you?